### PR TITLE
Fix CropAndResize op

### PR DESCRIPTION
**Pull request by dkurt**
Prevent floating point arithmetic influence on CropAndResize op

### CODE REVIEW

**Comment by yifeif** [comment]
Thanks @dkurt for the PR. Could you fill out the cla?

**Comment by dkurt** [comment]
@yifeif, sure. Please do not merge it immediately because I wanted to add a test.

**Comment by dkurt** [comment]
I signed it!

**Comment by dkurt** [comment]
@yifeif, it's ready to be reviewed. Thanks!

**Comment by tensorflowbutler** [comment]
Nagging Assignee @yifeif: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @yifeif: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @yifeif: It has been 45 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @yifeif: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @yifeif: It has been 74 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @yifeif: It has been 90 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by yifeif** [comment]
@jch1 do you mind taking a look? Thanks!

**Comment by tensorflowbutler** [comment]
Nagging Assignee @yifeif: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Review by kyscg — DISMISSED** [review]

File: tensorflow/core/kernels/crop_and_resize_op.cc [hunk 1]
```diff
@@ -250,22 +251,21 @@ struct CropAndResize<CPUDevice, T> {
             continue;
           }
           if (method_name == "bilinear") {
-            const int top_y_index = floorf(in_y);
-            const int bottom_y_index = ceilf(in_y);
+            const int bottom_y_index = std::min(top_y_index + 1, image_height - 1);
```
Comment (penpornk) [inline tensorflow/core/kernels/crop_and_resize_op.cc]: This will not work correctly if in_y is an integer, i.e., `in_y == top_y_index == bottom_y_index`.

**Review by penpornk — CHANGES_REQUESTED** [review]
I'm so sorry for our delay! Thank you for your patience! 
* I think there are build errors. Please check the failing tests, especially `Ubuntu CC`.
* Could you please explain what is wrong with the original code?

File: tensorflow/core/kernels/crop_and_resize_op.cc (hunk 1 above)
Comment (dkurt) [reply tensorflow/core/kernels/crop_and_resize_op.cc]: @penpornk, Thanks for review! Al this PR is driven by specific test case (see changed file). As far as I remember, without the changes in CropAndResize op it had last line with zeros.

Comment (dkurt) [reply tensorflow/core/kernels/crop_and_resize_op.cc]: Modified it. Let's check the test.

**Review by penpornk — CHANGES_REQUESTED** [review]
Got it! Thank you very much for the quick explanation and fixes! I think there is still some problems with the logic.

File: tensorflow/core/kernels/crop_and_resize_op.cc (hunk 1 above)
Comment (penpornk) [reply tensorflow/core/kernels/crop_and_resize_op.cc]: Thanks for the clarification! 
I reran the tests. 9 of them failed.

**Comment by rthadur** [comment]
@dkurt are you planning to address reviewer comments ?

**Comment by rthadur** [comment]
I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen!

**Pull request closed without merging**